Dart/src/com/jetbrains/lang/dart/ide/errorTreeView/DartProblemsFilter.java

```diff
@@ -109,6 +108,14 @@ public boolean include(@NotNull final DartProblem problem) {
       return false;
     }
 
+    if (myFileFilterMode == FileFilterMode.Directory && myCurrentFile != null) {
+      VirtualFile parent = myCurrentFile.getParent();
+      VirtualFile child = problem.getFile();
+      if (child != null && parent != null && !child.getPath().startsWith(parent.getPath())) {
```

Comment:
I'd add slash to avoid false positives:
`...startsWith(parent.getPath() + "/")`

BTW with this logic files in subdirectories are also included. Is it your intent?